src/main/java/com/aws/greengrass/deployment/DefaultDeploymentTask.java

```diff
@@ -194,6 +195,18 @@ public DeploymentResult call() throws InterruptedException {
         Optional<Set<String>> groupsForDeviceOpt;
         try {
             groupsForDeviceOpt = thingGroupHelper.listThingGroupsForDevice(retryCount);
+        } catch (GreengrassV2DataException e) {
+            if (e.statusCode() == 403) {
```

Comment:
Not blocking / follow-up checkin: Let's use constants rather than magic numbers (they should already exist).